### PR TITLE
feat: Add 2.0 activate, destroy, get, locate, query, revoke, set attr…

### PR DESCRIPTION
**Pull request by jskazinski**
- Add KMIP 2.0 operations for activate, destroy, get, locate, query, revoke, set attribute.
- Ran 'make' and all tests pass

### CODE REVIEW

File: kmip20/op_setattribute.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+// 6.1.47 Set Attribute
+
+// Table 296
+
+type SetAttributeRequestPayload struct {
+	UniqueIdentifier UniqueIdentifierValue
+	AttributeName    string
+	AttributeValue   string
```
Comment (ansel1) [inline kmip20/op_setattribute.go:16]: I don't think the request payload is supposed to directly contain AttributeName and AttributeValue.  From the spec (https://docs.oasis-open.org/kmip/kmip-spec/v2.0/os/kmip-spec-v2.0-os.html#_Toc6497573) and the test cases, it looks like it should be a structure (defined here: https://docs.oasis-open.org/kmip/kmip-spec/v2.0/os/kmip-spec-v2.0-os.html#_Toc6497524) which can contain any arbitrary attribute.  And from the test cases, it looks like the attributes can either directly be the attribute as a TTLV, or can be an `Attribute` structure.

So maybe this field can be defined like this:

```
    NewAttribute      Attributes  `ttlv:"DerivationData"`
```

**Comment by ansel1** [comment]
Also, please rebase on master.  Build is fixed.

File: kmip20/op_locate.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"context"
+
+	"github.com/gemalto/kmip-go"
+)
+
+// 6.1.27 Locate
+
+// Table 229
+
+type LocateRequestPayload struct {
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply kmip20/op_locate.go:13]: That is correct, we are not using these values for now, but will work on a separate later PR to expand and correct this type.

File: kmip20/op_setattribute.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+// Table 297
+
+type SetAttributeResponsePayload struct {
+	UniqueIdentifier string
+	AttributeName    string
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply kmip20/op_setattribute.go:23]: You are correct, I will update with the next commit.

File: kmip20/op_setattribute.go (hunk 1 above)
Comment (jskazinski) [reply kmip20/op_setattribute.go:16]: Thank you. I will review and make changes for the commit and review.

**Comment by ansel1** [comment]
Let me know when you are done and want me to look at it again.

File: kmip20/op_query.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	ObjectType               ObjectType
+	VendorIdentification     string
+	ServerInformation        string
+	ApplicationNamespace     string
+	ExtensionInformation     string
+	AttestationType          kmip14.AttestationType
+	RNGParameters            string
+	ProfileInformation       []ProfileName
+	ValidationInformation    []kmip14.ValidationAuthorityType
+	CapabilityInformation    []string
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply kmip20/op_query.go:31]: Added new CapabilityInformation to kmip20/op_query.go. Will add this, a few other types in the operation file, but perhaps the feedback will be that there is a better file for them, one that is common to other operation files.

File: kmip20/op_query.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	ClientRegistrationMethod kmip14.ClientRegistrationMethod
+	DefaultsInformation      string
+	ProtectionStorageMasks   string
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply kmip20/op_query.go:34]: We are working with 1 open-source server (github.com/OpenKMIP/PyKMIP) and two commercial servers. The two commercial servers we are working with only support up to version 1.4, so not helpful for the 2.0 items. I will what I can learn as far data returned.

[the remark this answers is not in the document]
Comment (jskazinski) [reply kmip20/op_query.go:34]: Based on Table 282: Query Request Payload in v2.1, it specifies a list, so I lean towards `ProtectectionStorageMasks []ProtectionStorageMask`

File: kmip20/op_revoke.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+	"github.com/gemalto/kmip-go"
+	"github.com/gemalto/kmip-go/kmip14"
+)
+
+// 6.1.40 Revoke
+
+// Table 269
+
+type RevocationReasonStruct struct {
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply kmip20/op_revoke.go:14]: I will submit this suggestion locally and commit.

**Comment by jskazinski** [comment]
I performed a rebase, made suggested changes, and pushed all the new files. I am seeing two concerns: 1) make lint fails, and fails in a clean clone of ithub.com/ThalesGroup/kmip-go.git and 2) the commit history contains more commits than just mine.

**Comment by jskazinski** [comment]
Done with this round of changes.

**Comment by jskazinski** [comment]
Automatic build tester reports a gofumpt issue with kmip20/op_get.go, but running  'make fmt' manually does not report any issues.

**Comment by ansel1** [comment]
I think your rebase didn't work or something.  I rebased and did a new pull request, #24 , which is now merged.

**Pull request closed without merging**